src/server/saved_objects/client/saved_objects_client.js

```diff
@@ -153,11 +153,25 @@ export class SavedObjectsClient {
       type: this._type,
       index: this._index,
       refresh: 'wait_for',
+      ignore: [404],
     });
 
-    if (response.found === false) {
+    const deleted = response.result === 'deleted';
+    if (deleted) {
+      return {};
+    }
+
+    // 404 might be because document is missing or index is missing,
+    // don't leak that implementation detail to the user
```

Comment:
Would be great if we could add an explanation here. (Basically answering the question: Why don't we want to leak this to the user?)

We've got this same comment in multiple places. Not sure if we should add an explanation in one place and "link" there, but it feels like we should make it clear here, at least.